mlrun/projects/project.py

```diff
@@ -2334,16 +2334,24 @@ def set_function(
         requirements: typing.Union[str, list[str]] = None,
         requirements_file: str = "",
     ) -> mlrun.runtimes.BaseRuntime:
-        """update or add a function object to the project
+        """
+        | Update or add a function object to the project.
+        | Function can be provided as an object (func) or a .py/.ipynb/.yaml URL.
 
-        function can be provided as an object (func) or a .py/.ipynb/.yaml url
-        support url prefixes::
+        | Creating a function from a single file is done by specifying ``func`` and disabling ``with_repo``.
+        | Creating a function with project source:
+        |   1. Specifying a relative ``func`` path and with_repo=True.
```

Comment:
|   1. Specify a relative ``func`` path and with_repo=True.
        |   2. Specify a module ``handler`` (e.g. ``handler=package.package.func``) without ``func``.